exe/kubernetes-deploy

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
+require 'kubernetes-deploy'
+
+options = {
+  namespace: ARGV[0],
+  context: ARGV[1],
+  environment: ENV['ENVIRONMENT'],
+  current_sha: ENV['REVISION'],
+  kubeconfig_path: ENV["KUBECONFIG"],
+}
+
+options[:template_dir] = ENV['K8S_TEMPLATE_FOLDER'] || File.expand_path("./config/deploy/#{options.fetch(:environment)}")
```

Comment:
Minimally, having an override available to point the script to another folder can useful when developing (e.g. you can run the script outside shipit and point it to a local folder). It also gives app devs flexibility in code organization; core might even need it? I'd say definitely keep this as an option.

That said, I'd be fine with moving this from an env var to a optional arg to the script if you think that's better. As of yesterday, the standard way to change the option will be via a key in your `shipit.yml`: 
```yaml
kubernetes:
  namespace: my-app-prod
  context: tier4
  template_dir: /app/config/templates
```
Right now, Shipit turns that into an env var [here](https://github.com/Shopify/shipit-engine/blob/master/app/models/shipit/deploy_spec/kubernetes_discovery.rb#L14). It is also possible to set the env var via your shipit machine env (would have been the only way until the first-class api), but I don't think preserving that is necessary.

If we keep this as an env var, making its name less explicit seems like an unnecessary 🚲 🏠  to me.